### PR TITLE
【Hackathon 4th No.15】Add GaussianNLLLoss rfc file.

### PR DESCRIPTION
**Pull request by Atlantisming**
Add GaussianNLLLoss rfc file.

代码及英文文档：https://github.com/PaddlePaddle/Paddle/pull/50843
中文文档：https://github.com/PaddlePaddle/docs/pull/5623

### CODE REVIEW

File: rfcs/APIs/20230222_api_design_for_gaussiannllloss.md [hunk 1]
```diff
@@ -0,0 +1,180 @@
...
+
+# 六、测试和验收的考量
+
+numpy没有gaussian_nll_loss的实现，所以由自己生成的函数进行前向测试和验收：
+
+1. 测试 API 在动态图和静态图下与 numpy 的一致性。
+2. 测试 CPU、GPU 上与 numpy 的一致性。
+3. 各reduction下计算一致
+4. 各参数输入有效。
+
```
Comment (GGBond8488) [inline rfcs/APIs/20230222_api_design_for_gaussiannllloss.md:171]: 补充可行性分析和排期规划

File: rfcs/APIs/20230222_api_design_for_gaussiannllloss.md [hunk 2]
```diff
@@ -0,0 +1,180 @@
...
+| 提交时间<input type="checkbox" class="rowselector hidden">   | 2023-02-22                                     |
+| 版本号                                                       | V1.0                                           |
+| 依赖飞桨版本<input type="checkbox" class="rowselector hidden"> | Develop                                        |
+| 文件名                                                       | 20230221_api_design_for_gaussiannllloss.md<br> |
+
+
+# 一、概述
+
+## 1、相关背景
+
```
Comment (GGBond8488) [inline rfcs/APIs/20230222_api_design_for_gaussiannllloss.md:15]: 补充一下loss计算公式

File: rfcs/APIs/20230222_api_design_for_gaussiannllloss.md [hunk 3]
```diff
@@ -0,0 +1,180 @@
+# xlogy 设计文档
```
Comment (GGBond8488) [inline rfcs/APIs/20230222_api_design_for_gaussiannllloss.md:1]: 变更为paddle.nn.gaussian_nll_loss 设计文档

File: rfcs/APIs/20230222_api_design_for_gaussiannllloss.md [hunk 4]
```diff
@@ -0,0 +1,180 @@
...
+
+`paddle.nn.functional.gaussian_nll_loss(input,
+    target,
+    var,
+    full=None,
+    eps=1e-6,
+    reduction: str="mean",
+    name:str=None,
+) -> Tensor:`
+ - Input(Tensor): `(N, *)` 或 `(*)` 其中 `*`表示任何数量的额外维度
```
Comment (GGBond8488) [inline rfcs/APIs/20230222_api_design_for_gaussiannllloss.md:124]: 补充各个参数代表的实际意义

File: rfcs/APIs/20230222_api_design_for_gaussiannllloss.md (hunk 3 above)
Comment (Atlantisming) [reply rfcs/APIs/20230222_api_design_for_gaussiannllloss.md]: 已修改

File: rfcs/APIs/20230222_api_design_for_gaussiannllloss.md (hunk 4 above)
Comment (Atlantisming) [reply rfcs/APIs/20230222_api_design_for_gaussiannllloss.md]: 已补充

File: rfcs/APIs/20230222_api_design_for_gaussiannllloss.md [hunk 5]
```diff
@@ -0,0 +1,189 @@
...
+2. 计算
+
+   1. 判断var是否小于eps
+   2. 计算loss
+
+
+3. 根据 `reduction`，输出 loss（同其余 functional loss 中的实现）
+
+
+# 六、测试和验收的考量
```
Comment (GGBond8488) [inline rfcs/APIs/20230222_api_design_for_gaussiannllloss.md:169]: 这里增加对不同数据类型的测试，明确该 loss 支持的输入数据类型和输出数据类型

**Review by PaddlePaddle-Gardener — APPROVED** [review]
LGTM

**Review by GGBond8488 — APPROVED** [review]
LGTM

**Review by luotao1 — APPROVED** [review]
